util/src/filesystem.cpp

```diff
@@ -15,8 +18,34 @@ std::string binaryPathToInstallDir(const char* path)
   // of any symbolic links and relative references.
   // The path is then parented twice (from binary to "bin" folder, and then
   // the root).
-  return fs::canonical(fs::system_complete(fs::path(path)))
-    .parent_path().parent_path().string();
+  // If the path does not exist, the given binary is searched for in the PATH
+  // environment variable.
+
+  if (fs::exists(fs::system_complete(fs::path(path))))
+  {
+    return fs::canonical(fs::system_complete(fs::path(path)))
+      .parent_path().parent_path().string();
+  }
+  else
+  {
+    std::string webserverPath;
+    std::string fullPath = std::getenv("PATH");
+    std::string delimiter = ":";
+
+    size_t pos = 0;
+    while ((pos = fullPath.find(delimiter)) != std::string::npos)
+    {
+      webserverPath = fullPath.substr(0, pos) + "/" + path;
+
+      if (fs::exists(fs::path(webserverPath)))
+      {
+        return fs::canonical(fs::path(webserverPath))
+          .parent_path().parent_path().string();
+      }
+
+      fullPath.erase(0, pos + delimiter.length());
+    }
+  }
```

Comment:
If no proper prefix is found in `PATH`, then there is no value returned from the function. Although it shall *never* occur according to the program workflow, an undefined behavior like that should still be avoided. Throwing an exception (e.g. `runtime_error`) could be a solution.